clang/lib/CodeGen/CGBuiltin.cpp

```diff
@@ -4674,9 +4674,10 @@ RValue CodeGenFunction::EmitBuiltinExpr(const GlobalDecl GD, unsigned BuiltinID,
   case Builtin::BI__builtin_cheri_address_set: {
     Value *Cap = EmitScalarExpr(E->getArg(0));
     Value *Address = EmitScalarExpr(E->getArg(1));
-    return RValue::get(
-        Builder.CreateIntrinsic(llvm::Intrinsic::cheri_cap_address_set,
-                                {IntPtrTy}, {EmitCastToVoidPtr(Cap), Address}));
+    return RValue::get(Builder.CreateBitCast(
```

Comment:
I don't think you want these bitcasts? They weren't there before. This is likely the cause of the type mismatch crashes.

Reply:
Yeah I've fixed that locally and thought I'd pushed it. Will update the pr tomorrow morning